node/builder_chain.go

```diff
@@ -240,7 +247,7 @@ func ConfigFullNode(c interface{}) Option {
 		// If the Eth JSON-RPC is enabled, enable storing events at the ChainStore.
 		// This is the case even if real-time and historic filtering are disabled,
 		// as it enables us to serve logs in eth_getTransactionReceipt.
-		If(cfg.Fevm.EnableEthRPC || cfg.Events.EnableActorEventsAPI, Override(StoreEventsKey, modules.EnableStoringEvents)),
+		If(cfg.Fevm.EnableEthRPC || cfg.Events.EnableActorEventsAPI || cfg.ChainIndexer.EnableIndexer, Override(StoreEventsKey, modules.EnableStoringEvents)),
```

Comment:
Maybe we don't want this one in here. Because I believe that as the code is right now, you can run ChainIndexer without events or eth enabled and it will work fine, indexing your messages. There is a `IsStoringEvents()` check in the tipset indexing code already. With this change, `IsStoringEvents()` will always be true.

So right now, there is a world in which you can run ChainIndexer just because you want faster message look-up.

Something we should discuss I think. Because enabling storing events does bloat your blockstore and slow tipset execution down a bit, so it's not free.

Reply:
Happy to discuss more. 

But -> I have yet to see a single use case for partial indexing. Not one user has asked for it/complained about it -> nada.

However, it is very easy to turn Indexing on thinking it will index everything(reasonable expectation) only to realise later that you should have turned on event storage as well -> but too late -> now you gotta backfill -> but because you have not stored events in the chain state -> sorry -> gotta re-compute/re-execute all those tipsets again to generate the events.

Reply:
I am happy to change this only if we have some RPC provider ask for this. Otherwise, we can change it when a user ask arises. Removing the Index DB is much easier than re-computing every tipset to regenerate the events.

Reply:
I can mostly get behind this attitude, but I'm concerned about the blinkered focus on "RPC providers"; I made this point to @BigLep in the doc PR too. With features like PDP, we're going to have some SPs needing to consume Eth APIs from their Lotus nodes, for example. They are not exposing their Lotus nodes to anyone else but they're going to need some of the features in here. There's more than just the SP bucket and the "RPC provider" bucket of users.

Before merging this I think we should properly document all of the optionality available and what the various options are going to make your node do or not do (such as turning on event AMT persistence in your blockstore if you enable ChainIndexer), and have a broader discussion about whether it's the right trade-off to reduce the number of buckets we can quantise our userbase into.